### ISSUE TITLE
Impossible to use `Attributes` in span because of Scala error: `Ambiguous overload. The overloaded alternatives of method apply in object Attributes with types`

### ISSUE THREAD

**Issue by jivanic-demystdata** [zio/zio-telemetry#911]
```scala
ZIO.service[Tracing].flatMap { tracing =>
  import tracing.aspects.*

  ZIO.attempt { ... ) @@ root(
    "my-trace", 
    attributes = Attributes(
      Attribute(AttributeKey.stringKey("id"), "1"),
      Attribute(AttributeKey.stringKey("example"), "value")
    )
}
```
fails to compile with:
```scala
Ambiguous overload. The overloaded alternatives of method apply in object Attributes with types
 [T, U]
  (attribute1: zio.telemetry.opentelemetry.common.Attribute[T], attribute2:
    zio.telemetry.opentelemetry.common.Attribute[U]):
    io.opentelemetry.api.common.Attributes
 [T]
  (attributes: zio.telemetry.opentelemetry.common.Attribute[T]*):
    io.opentelemetry.api.common.Attributes
both match arguments (zio.telemetry.opentelemetry.common.Attribute[String], zio.telemetry.opentelemetry.common.Attribute[String]) [89:28]
```

**Comment by grouzen** [comment]
Thanks for reporting!

**Comment by grouzen** [comment]
@jivanic-demystdata What scala version are you using? 

Also, could you try using the `Attribute.string` helper method (https://github.com/zio/zio-telemetry/blob/series/2.x/opentelemetry/src/main/scala/zio/telemetry/opentelemetry/common/Attribute.scala#L20) instead of wrapping a Java method `AttributerKey.stringKey`?

**Comment by jivanic-demystdata** [comment]
@grouzen I'm using Scala 3.5.2

Same error when using `Attribute.string`:
```scala
Ambiguous overload. The overloaded alternatives of method apply in object Attributes with types
 [T, U]
  (attribute1: zio.telemetry.opentelemetry.common.Attribute[T], attribute2:
    zio.telemetry.opentelemetry.common.Attribute[U]):
    io.opentelemetry.api.common.Attributes
 [T]
  (attributes: zio.telemetry.opentelemetry.common.Attribute[T]*):
    io.opentelemetry.api.common.Attributes
both match arguments (zio.telemetry.opentelemetry.common.Attribute[String], zio.telemetry.opentelemetry.common.Attribute[String]) [88:28]
```

**Comment by grouzen** [comment]
It seems working for the latest scala 2.13. Scala 3 apparently is more picky about this type of method resolution.

**Comment by grouzen** [comment]
@jivanic-demystdata Fixed in [3.1.0 release](https://github.com/zio/zio-telemetry/releases/tag/v3.1.0)
